src/Symfony/Component/PropertyAccess/StringUtil.php

```diff
@@ -197,7 +202,7 @@ public static function singularify($plural)
         }
 
         // Convert teeth to tooth, feet to foot
-        if (false !== ($pos = strpos($plural, 'ee')) && strlen($plural) > 3) {
+        if (false !== ($pos = strpos($plural, 'ee')) && strlen($plural) > 3 ! in_array($plural, self::$allowedEes)) {
```

Comment:
and your in_array check is broken because it is case sensitive so you don't always match it.